robottelo/ui/computeresource.py

```diff
@@ -441,3 +442,23 @@ def set_profile_values(self, resource_name, profile_name, **kwargs):
             self, resource_type)
         resource_profile_form.set_values(**kwargs)
         resource_profile_form.submit()
+
+    def check_image_os(self, os_name):
+        """Check if the OS is present, if not create the required OS
+
+        :param os_name: OS name to check, and create
+        :return Created os
+        """
+        # Check if OS that image needs is present or no, If not create the OS
+        result = entities.OperatingSystem().search(query={
+            u'search': u'title="{0}"'.format(os_name)
+        })
+        os = entities.OperatingSystem().create()
+        if len(result) == 0 or (os_name != result[0].title):
+            os = entities.OperatingSystem(
+                id=os.id,
+                name=os_name.split(' ')[0],
+                major=os_name.split(' ')[1].split('.')[0],
+                minor=os_name.split(' ')[1].split('.')[1],
+            ).update(['name', 'major', 'minor'])
+        return os
```

Comment:
Too much over-head for 3 tests having a cleanup fucn, will into @ldjebran approach without deleting the os.